### PR TITLE
[VIRTS-3453] Single instance of Chrome

### PR DESCRIPTION
**Pull request by djlawren**
## Description

These changes refactor the web driver related code to keep only one instance of Chrome open. This is done by changing the `WebDriverHelper` class to a Singleton that opens Chrome in its constructor. The same instance is then used by all web related workflows. They also make it so that only web related workflows need to contain a `WebDriverHelper`. Finally, signal handling was added to the main script to close Chrome when a kill signal is sent to the human since closing is not longer handled by a context manager.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

I ran pyhuman will all existing workflows on both a a Windows machine with Chrome installed and a Linux machine without Chrome installed. The Windows machine made use of all four workflows successfully, while the Linux machine only made use of the two workflows that do not require Chrome.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pyhuman/app/workflows/browse_web.py [hunk 1]
```diff
@@ -23,9 +24,8 @@ def action(self, extra=None):
     def _web_browse(self):
         random_website = self._get_random_website()
         try:
-            with self.driver as d:
-                d.get('https://' + random_website)
-                sleep(2)
+            self.driver.driver.get('https://' + random_website)
+            sleep(2)
```
Comment (mkultraWasHere) [inline pyhuman/app/workflows/browse_web.py]: Would it be beneficial to have the sleep timer a configurable parameter?

(if not, define the static value in a global at the top of module)

File: pyhuman/app/workflows/search_web.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
 from time import sleep
```
Comment (mkultraWasHere) [inline pyhuman/app/workflows/search_web.py]: Should we call this workflow 'google_search' or something different.?

I infer the same functionality from the workflows "search_web.py" and "browse_web.py"

File: pyhuman/app/workflows/search_web.py [hunk 3]
```diff
@@ -19,15 +21,14 @@ def action(self, extra=None):
...
-            elem.clear()
-            sleep(2)
-            elem.send_keys(search_string)
-            sleep(2)
-            elem.send_keys(Keys.RETURN)
+        self.driver.driver.get('https://www.google.com')
+        assert 'Google' in self.driver.driver.title
+        elem = self.driver.driver.find_element_by_name('q')
+        elem.clear()
+        sleep(2)
```
Comment (mkultraWasHere) [inline pyhuman/app/workflows/search_web.py]: Same comment here about the sleep variable.

File: pyhuman/app/utility/base_driver.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from abc import abstractmethod
+
+
+class Singleton(type):
```
Comment (mkultraWasHere) [inline pyhuman/app/utility/base_driver.py:4]: Like this.

**Review by mkultraWasHere — CHANGES_REQUESTED** [review]

**Review by mkultraWasHere — APPROVED** [review]

**Review by mkultraWasHere — CHANGES_REQUESTED** [review]
@djlawren - Woops, jumped the gun. Just noticed a loading issue on local testing. Should line 47 of `/app/human_svc.py` be updated as looks like every workflow `load` function call is getting "driver" parameter when not needed anymore?

**Review by mkultraWasHere — DISMISSED** [review]

File: app/human_svc.py [hunk 5]
```diff
@@ -42,12 +42,14 @@ async def load_available_workflows(self):
...
         try:
-            loaded = getattr(import_module(module), 'load')(driver=None)
-            await self.data_svc.store(Workflow(name=loaded.name, description=loaded.description, file=workflow_file))
+            module = import_module(module_path)
+            workflow_name = getattr(module, 'WORKFLOW_NAME')
+            workflow_description = getattr(module, 'WORKFLOW_DESCRIPTION')
+            await self.data_svc.store(Workflow(name=workflow_name, description=workflow_description, file=workflow_file))
         except Exception as e:
-            self.log.error('Error loading extension=%s, %s' % (module, e))
+            self.log.error('Error loading extension=%s, %s' % (module_path, e))
```
Comment (mchan143) [inline app/human_svc.py:52]: Tested this in my Ubuntu VM this morning and got the errors below when standing up the CALDERA server. Adding `webdriver_manager` to the requirements.txt should resolve, but did this come up for anyone else? 

```
2022-01-12 08:52:33 - ERROR (human_svc.py:52 _load_workflow_module) Error loading extension=plugins.human.pyhuman.app.workflows.browse_web, No module named 'webdriver_manager'
2022-01-12 08:52:33 - ERROR (human_svc.py:52 _load_workflow_module) Error loading extension=plugins.human.pyhuman.app.workflows.google_search, No module named 'webdriver_manager'
```

**Comment by mchan143** [comment]
After pip installing `webdriver-manager` to resolve the issue I mentioned above, everything else looked great! It should probably be added with a version to human's requirements.txt to get around the plugin load errors. Good work here.

**Review by mkultraWasHere — APPROVED** [review]

**Pull request merged**